crates/rpc/src/v03/method/simulate_transaction.rs

```diff
@@ -0,0 +1,456 @@
+use crate::{
+    cairo::ext_py::{
+        types::{FeeEstimate, FunctionInvocation, TransactionSimulation, TransactionTrace},
+        CallFailure,
+    },
+    context::RpcContext,
+    v02::{
+        method::call::FunctionCall,
+        types::{reply, request::BroadcastedTransaction},
+    },
+};
+
+use anyhow::anyhow;
+use pathfinder_common::{BlockId, CallParam, EntryPoint};
+use serde::{Deserialize, Serialize};
+use stark_hash::Felt;
+
+use super::common::prepare_handle_and_block;
+
+#[derive(Deserialize, Debug)]
+pub struct SimulateTrasactionInput {
+    block_id: BlockId,
+    transactions: Vec<BroadcastedTransaction>,
+    simulation_flags: dto::SimulationFlags,
+}
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+pub struct SimulateTransactionResult(pub Vec<dto::SimulatedTransaction>);
+
+crate::error::generate_rpc_error_subset!(
+    SimulateTransactionError: BlockNotFound,
+    ContractNotFound,
+    ContractError
+);
+
+impl From<CallFailure> for SimulateTransactionError {
+    fn from(value: CallFailure) -> Self {
+        match value {
+            CallFailure::NoSuchBlock => Self::BlockNotFound,
+            CallFailure::NoSuchContract => Self::ContractNotFound,
+            CallFailure::InvalidEntryPoint => Self::ContractError,
+            CallFailure::ExecutionFailed(e) => Self::Internal(anyhow!("Execution failed: {e:?}")),
+            CallFailure::Internal(e) => Self::Internal(anyhow!("Internal error: {e:?}")),
```

Comment:
I think we shouldn't return `CallFailure::Internal` error details over JSON-RPC. Neither `call` or `estimateFee` does this, because the strings inside this `Internal` are pretty much useless for end-users (typically internal stuff like serialization issues or inconsistencies in the inter-process commmunication with the Python subprocess).

Reply:
Done